aea/protocols/base/abstract_handler.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module defines the AbstractHandler class."""
+
+from abc import ABC, abstractmethod
+
+from aea.mail.base import Envelope
+
+
+class AbstractHandler(ABC):
+    """This is an interface for a message handler."""
+
+    # TODO: should we have the agent be accessible by the handlers?
+    # def __init__(self, agent):
+    #     """
+    #     Initialize the handler.
+    #
+    #     :param agent: the agent.
+    #     """
+    #     self.agent = agent
```

Comment:
It would probably be saver not to. If you recall the principle of object capability then  we don't want the handler to control the entire agent. At most we want it to be able to change (parts of) the agent's state.

Reply:
agreed.